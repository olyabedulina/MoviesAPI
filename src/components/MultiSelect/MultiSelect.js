import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.pcss'
import Checkbox from '../Checkbox'
import Dropdown from '../Dropdown'

const MultiSelect = ({
  placeholder,
  items,
  selectedItems,
  onChange
}) => {

    function handleItemChange(checked, itemValue) {
        const nextSelectedItems = checked ?
            selectedItems.concat(itemValue) :
            selectedItems.filter((value) => (value !== itemValue))

        onChange(nextSelectedItems)
    }

    const selectedItemTitles = selectedItems.map(
        (selectedItemValue) => items.find(
            ({ value }) => (value === selectedItemValue)
        ).title
    )

  return <Dropdown placeholder={`${selectedItemTitles.length ? selectedItemTitles.join(', ') : placeholder}`}>
      {
        items.map(({ value, title }) => <div key={value} className={CM.item}>
          <Checkbox
            checked={selectedItems.includes(value)}
            data={value}
            onChange={handleItemChange}
          >{title}</Checkbox>
        </div>)
      }
  </Dropdown>
}

MultiSelect.propTypes = {
    placeholder: PropTypes.string,
    items: PropTypes.array,
    selectedItems: PropTypes.array,
    onChange: PropTypes.func
};

export default MultiSelect
