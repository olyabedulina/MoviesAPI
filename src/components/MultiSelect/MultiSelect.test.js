import React from 'react';
import renderer from 'react-test-renderer';

import MultiSelect from './MultiSelect';

describe('Components :: Multiselect snapshots:', () => {
    it('Test :: "Multiselect component', () => {
        const array1 = [
            {
                id: '1',
                name: 'foo',
                isSelected: true,
                isIncludedInFilter: true
            },
            {
                id: '2',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            },
            {
                id: '3',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            }];

        const array2 = ['1']

        const actual = renderer
            .create(<MultiSelect
                placeholder="Select genre"
                items={array1}
                selectedItems={array2}
                onChange={() => {}}
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Multiselect component', () => {
        const array1 = [
            {
                id: '1',
                name: 'foo',
                isSelected: true,
                isIncludedInFilter: true
            },
            {
                id: '2',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            },
            {
                id: '3',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            }];

        const array2 = ['1']

        const actual = renderer
            .create(<MultiSelect
                items={array1}
                selectedItems={array2}
                onChange={() => {}}
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Multiselect component', () => {
        const array1 = [
            {
                id: '1',
                name: 'foo',
                isSelected: true,
                isIncludedInFilter: true
            },
            {
                id: '2',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            },
            {
                id: '3',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            }];

        const array2 = ['1', '2']

        const actual = renderer
            .create(<MultiSelect
                items={array1}
                selectedItems={array2}
                onChange={() => { alert(1)}}
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: "Multiselect component', () => {
        const array1 = [
            {
                id: '1',
                name: 'foo',
                isSelected: true,
                isIncludedInFilter: true
            },
            {
                id: '2',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            },
            {
                id: '3',
                name: 'foo',
                isSelected: false,
                isIncludedInFilter: true
            }];

        const array2 = []

        const actual = renderer
            .create(<MultiSelect
                placeholder="foo"
                items={array1}
                selectedItems={array2}
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });
});
