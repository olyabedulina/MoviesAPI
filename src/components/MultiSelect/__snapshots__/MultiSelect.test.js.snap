// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components :: Multiselect snapshots: Test :: "Multiselect component 1`] = `
<div
  className="  dropdown "
>
  <div
    className="dropdownInput"
  >
    <div
      className="dropdownPlaceholder"
    >
      foo
    </div>
    <div
      className="dropdownInputButton dropdownInputButtonUp"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Components :: Multiselect snapshots: Test :: "Multiselect component 2`] = `
<div
  className="  dropdown "
>
  <div
    className="dropdownInput"
  >
    <div
      className="dropdownPlaceholder"
    >
      foo
    </div>
    <div
      className="dropdownInputButton dropdownInputButtonUp"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Components :: Multiselect snapshots: Test :: "Multiselect component 3`] = `
<div
  className="  dropdown "
>
  <div
    className="dropdownInput"
  >
    <div
      className="dropdownPlaceholder"
    >
      foo, foo
    </div>
    <div
      className="dropdownInputButton dropdownInputButtonUp"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Components :: Multiselect snapshots: Test :: "Multiselect component 4`] = `
<div
  className="  dropdown "
>
  <div
    className="dropdownInput"
  >
    <div
      className="dropdownPlaceholder"
    >
      foo
    </div>
    <div
      className="dropdownInputButton dropdownInputButtonUp"
      onClick={[Function]}
    />
  </div>
</div>
`;
